docs_src/microservices/application/BuiltIn.md

```diff
@@ -0,0 +1,256 @@
+    Only  operators that return true or false are supported. See http://jsonlogic.com/operations.html# for the complete list paying attention to return values. Any operator that returns manipulated data is currently not supported. For more advanced scenarios checkout [EMQ X Kuiper](https://github.com/emqx/kuiper).
+
+!!! tip
+    Leverage http://jsonlogic.com/play.html to get your rule right before implementing in code. JSON can be a bit tricky to get right in code with all the escaped double quotes.
+
+## Encryption
+There is one encryption transform included in the SDK that can be added to your pipeline. 
+
+| Factory Method                   | Description | 
+|----------------------------------|-------------|
+| NewEncryption(key string, initializationVector string) | This function returns a `Encryption` instance initialized with the passed in key and initialization vector. This `Encryption` instance is used to access the following encryption function that will use the specified key and initialization vector. |
+
+### AES
+`EncryptWithAES` - This function receives a either a `string`, `[]byte`, or `json.Marshaller` type and encrypts it using AES encryption and returns a `[]byte` to the pipeline.
+``` go
+NewEncryption("key", "initializationVector").EncryptWithAES
+```
+
+## Batch
+Included in the SDK is an in-memory batch function that will hold on to your data before continuing the pipeline. There are three functions provided for batching each with their own strategy.
+
+
+| Factory Method                   | Description | 
+|----------------------------------|-------------|
+|NewBatchByTime(timeInterval string) | This function returns a `BatchConfig` instance with time being the strategy that is used for determining when to release the batched data and continue the pipeline. `timeInterval` is the duration to wait (i.e. `10s`). The time begins after the first piece of data is received. If no data has been received no data will be sent forward. 
+``` go
+// Example: 
+NewBatchByTime("10s").Batch
+```
+| NewBatchByCount(batchThreshold int) | This function returns a `BatchConfig` instance with count being the strategy that is used for determining when to release the batched data and continue the pipeline. `batchThreshold` is how many events to hold on to (i.e. `25`). The count begins after the first piece of data is received and once the threshold is met, the batched data will continue forward and the counter will be reset.
+``` go
+// Example:
+NewBatchByCount(10).Batch
+```
+| NewBatchByTimeAndCount(timeInterval string, batchThreshold int) | This function returns a `BatchConfig` instance with a combination of both time and count being the strategy that is used for determining when to release the batched data and continue the pipeline. Whichever occurs first will trigger the data to continue and be reset.
+``` go
+// Example:
+NewBatchByCount("30s", 10).Batch
+```
+`Batch` - This function will apply the selected strategy in your pipeline.
```

Comment:
Should we add some type of warning here about batching too much and running out of memory?  I am not sure if we can quantify it, but batching to millions is probably not recommended?